crates/medea-reactive/src/field/mod.rs

```diff
@@ -91,27 +88,73 @@ where
     where
         F: Fn(&D) -> bool + 'static,
     {
+        // TODO: This is kinda broken.
+        //       1. Inner value is 0.
+        //       2. let fut = field.when_eq(0).
+        //       3. Change value to 1.
+        //       4. fut.await = Poll::Ready.
+        //       I suggest interior mutability + async fn's.
```

Comment:
So, what we gonna do with this TODO?

Reply:
@tyranron ,

Технически это не корректно, но ситуаций, когда бы мы поллили эту футуру не сразу у нас нет и не предвидится, а корректная реализация будет требовать существенных изменений в `medea-reactive`, поэтому было решено не делать это сейчас.

Reply:
@evdokimovs if os, do we need

```rust
if (assert_fn)(&self.data) {
            Box::pin(futures::future::ok(()))
```
branch at all?

May be it's better to always go with a second one?

Reply:
@tyranron ,

Эта часть достаточно важна, поскольку при всех использованиях этой функции нам нужно, чтобы футура зарезолвилась, если условие уже выполнено. Мы можем этого достигнуть если будем делать эту проверку вручную перед вызовом `when`, но тогда нам придется при каждом вызове `when` писать 
`if val.get() != foo { val.when(...).await }`, что выглядит не очень удобным и надежным. Тоже самое будет и с методом `when_eq`, поскольку он использует `when` под капотом.

Reply:
@tyranron ,

Еще есть ощущение, что мы все равно рано или поздно придем к тому, чтобы сделать все мутации внутренними, поскольку при реальном применении мы почти всегда используем `ObservableCell`, либо оборачиваем в `RefCell` в случае с коллекциями. Если мы это сделаем, то пофиксить это поведение будет легко.

Reply:
@evdokimovs ok then, please create an appropriate issue and mention it in the TODO-comment.

Otherwise it would be too easy to forget this and kill a lot of time debugging accidential problems why `not eq` is `eq`.

Reply:
@tyranron ,

Хочу дополнительно прояснить почему так.

> May be it's better to always go with a second one?

Второй бренч детектит изменения и только тогда триггерит проверки. Т.е. если условие удовлетворено в данный момент, то второй бренч не зарезолвится. Нам тут в любом случае нужно делать проверку при poll'е, и это или `async fn`, или нужно сам value завернуть в `Rc` и мувать его в результирующую футуру. Я за `async fn`, но так просто его сейчас не запилить, поскольку есть места где эта структура лежит за `RefCell`, ну а `borrow().my_fn().await` понятно чем чревато.

Reply:
@alexlapa well, seeing this fact and the necessitty of boxing everywhere in the library, I only ensure that the design of `futures-signals` is much more clever-minded.

Reply:
@tyranron ,

Нам нужны `'static` для [`future_to_promise`](https://rustwasm.github.io/wasm-bindgen/api/wasm_bindgen_futures/fn.future_to_promise.html) ¯\_(ツ)_/¯.